### DISCUSSION TITLE
Dependabot PRs warning

### DISCUSSION THREAD

**Discussion by natestemen** [unitaryfund/mitiq#1948] [General]
Starting in https://github.com/unitaryfund/mitiq/pull/1672, and then again in https://github.com/unitaryfund/mitiq/pull/1917 we have loosened the dependency specification from a single minor version (using `~=`) to a range of values (using `>=` and `<`). An exposition of the reasons for making this change can be found in https://github.com/unitaryfund/mitiq/issues/1667, but one of the main reasons is to make installs easier. The more permissive the dependency specification, the easier the install.

As an unforeseen consequence, dependabot is now less useful. Take this PR as an example https://github.com/unitaryfund/mitiq/pull/1891 which made the following change.

```diff
 numpy>=1.22.0
-scipy>=1.5.0,<=1.10.1
+scipy>=1.5.0,<=1.11.1
 cirq>=1.0.0,<1.3.0
```
This is exactly the kind of broadening that we want, but it backfires. When doing an install, `pip` can now find a version of `scipy` that satisfies the installation requirements, even if it is not in the `1.10.10<X<=1.11.1` range. Indeed in the build we can see `scipy 1.10.1` is installed. The builds then pass, but this does not provide any evidence that Mitiq works with `scipy 1.11.1`.

## what's next?

We can either do one of two things

1. Revert to tighter specifications, and more useful dependabot PRs
2. Do more manual testing on dependabot PRs (sometimes this should be simple like double checking that the PR is actually using the latest version, and with other where it's not, it will require is attempting to build mitiq locally with the bumped version).

I don't love either option, but have a slight preference to try 2, at least for now.

**Comment by andreamari** [comment]
I agree with manual testing for the 3 packages in `requirements.txt`. All packages in `dev_requirements.txt` are tight, so dependabot should work well for them.

**Reply by natestemen** [reply]
That sounds good. I'll close this discussion out for now, but feel free to come back to it if there are future issues.
